packages/react/src/components/ActionCard/ActionCard.tsx

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Card from '@mui/material/Card';
+import CardActions from '@mui/material/CardActions';
+import CardContent from '@mui/material/CardContent';
+import {FC, ReactElement, ReactNode} from 'react';
+import Button from '../Button';
+import Typography from '../Typography';
+import {WithWrapperProps} from '../../models';
+
+export interface ActionCardProps {
+  /**
+   * The text to be displayed in the action button.
+   */
+  actionText: string;
+  /**
+   * The description of the card.
+   */
+  description: string;
+  /**
+   * The image to be displayed in the card.
+   */
+  image: ReactNode;
+  /**
+   * Callback method to be called when the action button is clicked.
+   */
+  onActionClick: () => void;
+  /**
+   * The title of the card.
+   */
+  title: string;
+}
+
+const COMPONENT_NAME: string = 'ActionCard';
+
+const ActionCard: FC<ActionCardProps> & WithWrapperProps = (props: ActionCardProps): ReactElement => {
+  const {image, title, description, actionText, onActionClick} = props;
```

Comment:
Shall we spread `rest` in the root `Card` component.
```suggestion
  const {image, title, description, actionText, onActionClick, ...rest} = props;
```